src/test/scala/com/fulcrumgenomics/vcf/MakeMixtureVcfTest.scala

```diff
@@ -0,0 +1,239 @@
+    val mixer   = new MakeMixtureVcf(input=dummyVcf, output=dummyVcf, samples=Seq("s1@0.5", "s2", "s3"), outputSampleName="mix")
+    val ss      = mixer.determineSamples(builder.header)
+    ss shouldBe Seq(Sample("s1", 0.5), Sample("s2", 0.25), Sample("s3", 0.25))
+  }
+
+  it should "throw an exception if a sample is specified that's not in the VCF" in {
+    val samples = Seq("s1", "s2", "s3", "s4")
+    val builder = new VariantContextSetBuilder(samples)
+    val mixer   = new MakeMixtureVcf(input=dummyVcf, output=dummyVcf, samples=Seq("s1", "s2", "s7"), outputSampleName="mix")
+    an[Exception] should be thrownBy { mixer.determineSamples(builder.header) }
+  }
+
+  it should "throw an exception if the proportions add up to less than 1" in {
+    val samples = Seq("s1", "s2", "s3", "s4")
+    val builder = new VariantContextSetBuilder(samples)
+    val mixer   = new MakeMixtureVcf(input=dummyVcf, output=dummyVcf, samples=Seq("s1@0.25", "s2@0.25"), outputSampleName="mix")
+    an[Exception] should be thrownBy { mixer.determineSamples(builder.header) }
+  }
+
+  it should "throw an exception if the proportions add up to more than 1" in {
+    val samples = Seq("s1", "s2", "s3", "s4")
+    val builder = new VariantContextSetBuilder(samples)
+    val mixer   = new MakeMixtureVcf(input=dummyVcf, output=dummyVcf, samples=Seq("s1@0.75", "s2@0.75"), outputSampleName="mix")
+    an[Exception] should be thrownBy { mixer.determineSamples(builder.header) }
+  }
+
+  it should "throw an exception if the same sample is specified more than once" in {
+    val samples = Seq("s1", "s2", "s3", "s4")
+    val builder = new VariantContextSetBuilder(samples)
+    val mixer   = new MakeMixtureVcf(input=dummyVcf, output=dummyVcf, samples=Seq("s1@0.5", "s1@0.5"), outputSampleName="mix")
+    an[Exception] should be thrownBy { mixer.determineSamples(builder.header) }
+  }
+
+  "MakeMixtureVcf.updateAlleleFractionsForSample" should "extract simple fractions when no AF field is used" in {
+    val samples = Seq("s1", "s2")
+    val builder = new VariantContextSetBuilder(samples)
+    builder.addVariant(start=10, sampleName=Some("s1"), variantAlleles=List("A", "C"), genotypeAlleles=List("A"))
+    builder.addVariant(start=10, sampleName=Some("s2"), variantAlleles=List("A", "C"), genotypeAlleles=List("A", "C"))
+    val ctx = builder.iterator.next()
+
+    val mixer     = new MakeMixtureVcf(input=dummyVcf, output=dummyVcf, outputSampleName="mix")
+    val fractions = mutable.Map[Allele,Double](ctx.getAlleles.map(a => a -> 0d).toSeq:_*)
+
+    mixer.updateAlleleFractionsForSample(ctx, Sample("s1", 0.5), fractions)
+    fractions(ctx.getAllele("A")) shouldBe 0.5
+    fractions(ctx.getAllele("C")) shouldBe 0.0
+
+    mixer.updateAlleleFractionsForSample(ctx, Sample("s2", 0.5), fractions)
+    fractions(ctx.getAllele("A")) shouldBe 0.75
+    fractions(ctx.getAllele("C")) shouldBe 0.25
+  }
+
+  it should s"extract a single-valued AF from a field" in {
```

Comment:
At one point I was looping over that test with different inputs, then I changed my mind.